server/container/guice/cassandra-guice/src/main/java/org/apache/james/server/CassandraCleanupProbe.java

```diff
@@ -17,33 +17,24 @@
  * under the License.                                           *
  ****************************************************************/
 
-package org.apache.james.modules;
+package org.apache.james.server;
 
-import org.apache.james.backends.es.EmbeddedElasticSearch;
-import org.apache.james.mailbox.extractor.TextExtractor;
-import org.apache.james.mailbox.store.extractor.DefaultTextExtractor;
-import org.apache.james.util.Host;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
 
-import com.google.inject.AbstractModule;
+import org.apache.james.backends.cassandra.init.CassandraTableManager;
+import org.apache.james.utils.GuiceProbe;
 
-public class CassandraJmapServerModule extends AbstractModule {
+public class CassandraCleanupProbe implements GuiceProbe {
+    private final CassandraTableManager tableManager;
 
-    private static final int LIMIT_TO_3_MESSAGES = 3;
-    private final EmbeddedElasticSearch embeddedElasticSearch;
-    private final Host cassandraHost;
-
-    public CassandraJmapServerModule(EmbeddedElasticSearch embeddedElasticSearch, Host cassandraHost) {
-        this.embeddedElasticSearch = embeddedElasticSearch;
-        this.cassandraHost = cassandraHost;
+    @Inject
+    public CassandraCleanupProbe(CassandraTableManager tableManager) {
+        this.tableManager = tableManager;
     }
 
-    @Override
-    protected void configure() {
-        install(new CassandraTestModule(cassandraHost));
-        install(new TestElasticSearchModule(embeddedElasticSearch));
-        install(new TestJMAPServerModule(LIMIT_TO_3_MESSAGES));
-
-        install(binder -> binder.bind(TextExtractor.class).to(DefaultTextExtractor.class));
+    @PreDestroy
```

Comment:
how are you handling the call ordering ? What if there's a lifecycle depending on a cassandra table ? what if TableManager is already "closed" ?

Reply:
It doesn't in my understanding (or maybe just by accident). You use multibind to register your Probe, it uses a Set and AFAIK doesn't guarantee any ordering. How do you know it will respect the order you want ? I may I missed something ....

Reply:
The probe is staged after init, because the GuiceProbeProvider is recovered after performing init.

With the stager, it's not the inject which counts but the actual init <3